synapse/util/caches/deferred_cache.py

```diff
@@ -119,21 +128,27 @@ def check_thread(self):
     def get(
         self,
         key: KT,
-        default=_Sentinel.sentinel,
         callback: Optional[Callable[[], None]] = None,
         update_metrics: bool = True,
-    ):
+    ) -> defer.Deferred:
```

Comment:
I had debated making this async at some point, which should work if all callers just await it, which they might now that you added `get_immediate`?

Reply:
well, the only place this is called is https://github.com/matrix-org/synapse/pull/8572/files#diff-3aa4bc9d8830ffd035d53b8af86b548a21f2a4cddb3e5240445661614a88a08fR313, which doesn't await it. In any case the logcontext-wrangling means you end up needing to wrap it with `make_deferred_yieldable`.

I considered making `get()` do the `make_deferred_yieldable` call itself, but decided that the resultant asymmetry was confusing.